src/decorator.ts

```diff
@@ -0,0 +1,34 @@
+import { untilDestroyed } from './take-until-destroy';
+
+/**
+ * Autamatically unsubscribes from pipe when component destroyed.
+ *
+ * @example
+ * ```ts
+ * @Component({...})
+ * class MyComponent implements OnDestroy {
+ *   @AutoUnsubscribe()
+ *   stream$ = new Observable(...);
+ *
+ *   // OnDestroy method is required by Angular Compiler
+ *   ngOnDestroy() {}
+ * }
+ * ```
+ *
+ * Uses {@link untilDestroyed} operator on the pipe.
+ *
+ * Do not forget to implement {@link OnDestroy} life-cycle hook.
+ */
+export function AutoUnsubscribe(destroyMethodName?: string): PropertyDecorator {
+  return (target, propKey) => {
+    const originalProp = target[propKey];
+
+    if (delete target[propKey]) {
+      Object.defineProperty(target, propKey, {
+        enumerable: true,
+        configurable: true,
+        value: (originalProp as Observable<any>).pipe(untilDestroyed(target)),
```

Comment:
Agree that name is better to be `@WithUntilDestroyed`.

Decorator is better because:
1. You automatically capture `this` object
2. You don't need to write `.pipe` for it
3. Readability. Sometimes the `.pipe` chain is lengthy and you might forget or not see easily if there is a `takeUntil/untilDestroyed` operator applied. With decorator it's first thing you see.

Reply:
Name updated to `@WithUntilDestroyed`